spec/kitchen/driver/ec2_spec.rb

```diff
@@ -678,6 +678,16 @@
       include_examples "common create"
     end
 
+    context "with key pair configured to false" do
+      before do
+        config[:aws_ssh_key_id] = false
+        expect(driver).to receive(:submit_server).and_return(server)
```

Comment:
Maybe include a negative expectation for setting `aws_ssh_key_id` to `false`?

```
expect(driver).to_not receive(:create_key)
```

Looking at the spec, I found myself wondering what was different.

Reply:
The key won't get created if you provide it either